src/main/java/build/buildfarm/instance/memory/MemoryInstance.java

```diff
@@ -396,7 +397,13 @@ public boolean putOperation(Operation operation) throws InterruptedException {
             .setNanos(actionTimeout.getNanos() + delay.getNanos())
             .build();
         // this is an overuse of Watchdog, we will never pet it
-        Watchdog operationTimeoutDelay = new Watchdog(timeout, () -> expireOperation(operation));
+        Watchdog operationTimeoutDelay = new Watchdog(timeout, () -> {
+          try {
+            expireOperation(operation);
+          } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
Where is the interrupted flag tested?

Reply:
It isn't, and the param is already an InterruptingRunnable, reducing.